xmodule/docs/decisions/0004-restructure-video-block.rst

```diff
@@ -0,0 +1,101 @@
+
+Accessibility is a key focus area, and the new player aims to be compliant with the latest WCAG guidelines. This includes providing Audio Descriptions, focus state visibility, and other accessibility features, which are currently either lacking or not up to the mark in the existing system. The new architecture aims to rectify this by integrating these features into the core of the new Video Player.
+
+In summary, the context for this Architecture Decision Record (ADR) is introduction of the new Video XBlock aims to address multiple pain points, ranging from user experience and accessibility to extensibility and modernization of the tech stack. It represents a significant step forward in aligning the platform with modern web standards and user expectations, while also laying the groundwork for future enhancements.
+
+
+Decision
+--------
+
+1. **Move out current Video XBlock from xmodules and place it in a separate repository**
+
+   We have decided to extract the current Video XBlock from the xmodules directory and relocate it into its own dedicated repository within the github.com/openedx organization. This decision is driven by the need for improved modularity, maintainability, and the facilitation of continuous integration and deployment practices.
+   * A new repository will be created under the Open edX GitHub organization.
+   * The repository will adhere to Open edX's best practices, including code quality, testing, and documentation.
+   * The new XBlock will be hosted within the Open edX organization on GitHub, ensuring community involvement and adherence to Open edX standards.
+   * The proposed architecture for installing and using new xblock is the following:
+
+     .. image:: ./Architecture.png
+
+2. **React.js and Video.js Integration**
+   * React.js will be used to build a responsive, accessible frontend for a new Xblock as proposed in https://github.com/openedx/XBlock/issues/635 and prototyped in https://github.com/openedx/XBlock/issues/634.
+   * Video.js will be integrated for video playback, following the official guidelines (`Video.js Docs <https://docs.videojs.com/>`_).
+   * An NPM package will be created for the frontend, and it will be versioned and published (`Issue 635 <https://github.com/openedx/XBlock/issues/635>`_).
+
+3. **Feature Toggle in edx-platform**
+   * A feature toggle will be introduced using Django Waffle flags.
+   * This will allow for a manual rollout and the ability to quickly revert to the old player if issues arise.
+
+4. **Integration into Legacy Interface and Course Authoring**
+   * The new XBlock will be embedded in iframes in the legacy interface.
+   * In the Course Authoring MFE, the XBlock will be integrated natively using React.js components.
+
+5. **Migration Necessity**
+  * The data migration will not be required as long as the current video block backend is used.
+  * Data and metadata will be preserved during the migration.
```

Comment:
Are there any old/obscure/deprecated features of the existing VideoBlock that _won't_ make the transition?

Reply:
From https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3674734593/Approach+Memo+Technical+Discovery+React-based+Video+Player#Technical-Requirements we do have an item:

Discovery will need to be done on what [fields currently exist in the XBlock](https://github.com/openedx/edx-platform/blob/master/xmodule/video_block/video_xfields.py), and as much as possible should go through the Deprecation process before beginning the process of migrating any over

Such as the multi-speed YouTube upload option, which we used in 2012 before YT started generating multiple speeds for you. Open questions: Is anyone using video bumpers? What features weren’t ported to the MFE, or are no longer exposed in the UI? etc